### PR TITLE
Breaking Change: Remove built-in icons

### PR DESCRIPTION
**Pull request by reesscot**
#### Description of changes
This PR is an updated version of https://github.com/aws-amplify/amplify-ui/pull/1310. It removes all the icons not used in our code base and updates all the docs examples to use `pathData` instead of the built-in icons.

**`@aws-amplify/ui-react` BEFORE removing icons:**
```
npm notice === Tarball Details === 
npm notice name:          @aws-amplify/ui-react                   
npm notice version:       2.19.1                                  
npm notice filename:      @aws-amplify/ui-react-2.19.1.tgz        
npm notice package size:  773.5 kB                                
npm notice unpacked size: 6.2 MB                                  
npm notice shasum:        ac9a71590f1750b68f6b889a88b1b0a3347638e5
npm notice integrity:     sha512-MLYaghCnYK7Cv[...]rim+eNwHvwyKw==
npm notice total files:   3091                                    
npm notice 
```
**`@aws-amplify/ui-react` AFTER removing icons:**
```
npm notice === Tarball Details === 
npm notice name:          @aws-amplify/ui-react                   
npm notice version:       2.19.1                                  
npm notice filename:      @aws-amplify/ui-react-2.19.1.tgz        
npm notice package size:  195.0 kB                                
npm notice unpacked size: 2.1 MB                                  
npm notice shasum:        a88772137083cfaf30cc7cefc098847365643e4d
npm notice integrity:     sha512-zo0DYkzZCrLJh[...]7BsrC8c75Frww==
npm notice total files:   469                                     
npm notice 
```

#### Checklist

- [x] Add BREAKING CHANGE changeset
- [x] PR description included
- [ ] `yarn test` passes
- [ ] Tests are updated
- [x] No side effects or [`sideEffects`](https://github.com/aws-amplify/amplify-ui/blob/main/packages/react/CONTRIBUTING.md#code-standards) field updated
- [x] Relevant documentation is changed or added (and PR referenced)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by dbanksdesign — APPROVED** [review]
LGTM!

**Comment by ayush987goyal** [comment]
Can you please make sure the following icons are not removed since they are being used in the staging repo?

1. IconClose
2. IconCheckCircleOutline
3. IconHighlightOff
4. IconFiberManualRecord

**Comment by reesscot** [comment]
> * IconCheckCircleOutline
> 
> * IconHighlightOff
> 
> * IconFiberManualRecord

Thanks for the review @ayush987goyal. IconClose is still there but I will add the other three back:
IconCheckCircleOutline
IconHighlightOff
IconFiberManualRecord

**Comment by reesscot** [comment]
> > * IconCheckCircleOutline
> > * IconHighlightOff
> > * IconFiberManualRecord
> 
> Thanks for the review @ayush987goyal. IconClose is still there but I will add the other three back: IconCheckCircleOutline IconHighlightOff IconFiberManualRecord

Added back the icons needed by staging repo

**Comment by ayush987goyal** [comment]
Thanks @reesscot . Let's discuss offline to see how can we remove the dependency form staging repo. We can then remove these ones too later.

File: .changeset/three-garlics-remember.md [hunk 1]
```diff
@@ -0,0 +1,11 @@
+---
+"@aws-amplify/ui-react": major
+---
+
+🚨BREAKING CHANGE🚨: Built-in icons removed
+
+We're removed the majority of the built-in icons (Icon360, IconSave, etc) from the `@aws-amplify/ui-react` package for a few reasons:
```
Comment (joebuono) [inline .changeset/three-garlics-remember.md:7]: nit: "We've"

File: .changeset/three-garlics-remember.md [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+* Including icons makes the package large. While in most cases the icons can be tree-shaken out of the final application, by removing them we can ensure they are are never included.
```
Comment (joebuono) [inline .changeset/three-garlics-remember.md:8]: nit: “we can ensure they are ~~are~~ never included”

File: .changeset/three-garlics-remember.md [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+* There are plenty of other React icon libraries that are compatible with Amplify UI components
+
+We've left the `<Icon>` component, which still allows customers to easily add SVG icons using the `pathData` prop. A limited set of of the icons have also been left for internal use inside of Amplify UI primitives.
```
Comment (joebuono) [inline .changeset/three-garlics-remember.md:11]: nit: “A limited set of ~~of~~ the icons”

**Review by joebuono — APPROVED** [review]

File: .changeset/three-garlics-remember.md [hunk 4]
```diff
@@ -0,0 +1,12 @@
+---
+"@aws-amplify/ui-react": major
+---
+
+🚨BREAKING CHANGE🚨: Built-in icons removed
+
+We've removed the built-in icons (Icon360, IconSave, etc) from the `@aws-amplify/ui-react` package. This change was made for a few reasons:
+* There are many other React icon libraries that are compatible with Amplify UI components
+* Removing the icons reduces the overall package size by about 74%.
+* While in most cases the icons are tree-shaken out of the final app bundle, by removing the icons we ensure smaller bundles sizes for all users.
```
Comment (joebuono) [inline .changeset/three-garlics-remember.md:10]: nit: "smaller bundle sizes"

File: .changeset/three-garlics-remember.md [hunk 5]
```diff
@@ -0,0 +1,12 @@
...
+
+We are not removing the `<Icon>` component, which allows customers to easily add SVG icons using the `pathData` prop. A limited set of of the icons have been left for internal only use in Amplify UI primitives.
```
Comment (joebuono) [inline .changeset/three-garlics-remember.md:12]: "limited set of ~~of~~ the icons"

File: .changeset/three-garlics-remember.md [hunk 6]
```diff
@@ -0,0 +1,12 @@
+---
+"@aws-amplify/ui-react": major
+---
```
Comment (wlee221) [inline .changeset/three-garlics-remember.md:3]: 🎉

**Pull request merged**